content/shmem_init_thread.tex

```diff
@@ -0,0 +1,78 @@
+\apisummary{
+Initializes the \openshmem{} library, similar to SHMEM\_INIT(), and in addition
+perform the initialization required for thread-safe invocation of \openshmem{} functions.}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+int shmem_init_thread(int requested, int *provided);
+\end{Csynopsis}
+
+\begin{Fsynopsis}  
+INTEGER requested, provided
+     
+SHMEM_INIT_THREAD(requested, provided)
+\end{Fsynopsis}  
+
+\begin{apiarguments}
+\apiargument{IN}{requested}  {The thread level support requested by the user.
+The correct values are SHMEM\_THREAD\_SINGLE, SHMEM\_THREAD\_FUNNELED, or SHMEM\_THREAD\_MULTIPLE}
+\apiargument{OUT}{provided}{The thread level support provided by the \openshmem{} implementation.}
+\end{apiarguments}
+
+
+\apidescription{
+    \FUNC{shmem\_init\_thread}  initializes the \openshmem{} library similar to
+    \FUNC{shmem\_init}, and in addition perform the initialization required for
+    thread-safe invocation of \openshmem{} functions. The argument
+    \textit{requested} is used to specify the desired level of thread support.
+    The function returns the support provided by the library. There are four 
+    levels of threading support.
+ 
+ \begin{itemize}
+\item SHMEM\_THREAD\_SINGLE: The \openshmem{} program may not be multithreaded, 
+and all \openshmem{} interfaces are invoked by a single thread. 
+
+\item SHMEM\_THREAD\_FUNNELED: 
+The \openshmem{} program may be multithreaded, however, the 
+program must ensure that only one thread invokes the \openshmem{}
+interfaces.
+
+\item SHMEM\_THREAD\_SERIALIZED: 
+The \openshmem{} program may be multithreaded, however, the 
+program must ensure that only one thread invokes the \openshmem{}
+interfaces at any given instance i.e., the \openshmem{} interfaces 
+are not invoked concurrently by multiple threads.
+
+\item SHMEM\_THREAD\_MULTIPLE: 
+The \openshmem{} program may be multithreaded, and any 
+thread may invoke the \openshmem{} interfaces.
+
+\item SHMEM\_THREAD\_UNSUPPORTED: The \openshmem{} specification 
```

Comment:
> For example, we have argued on the WG calls that SHMEM_FUNNELED should allow any one thread (not just main thread) to make OpenSHMEM calls. However, we did not pursue because of the potential overheads. This doesn’t mean we should include this mode, or it is not important.

@manjugv, with SERIALIZED included in the list, doesn't this cover the case where one thread (not just the main thread) can make OpenSHMEM calls.